tests/test_contexts.py

```diff
@@ -35,29 +33,29 @@ def block_checks_glb(name, glbs, body, obs=None):
     block = glbs[name]
     obs = obs or {}
     for k, v in obs.items():
-        yield assert_equal, v, glbs[k]
+        assert v == glbs[k]
     if isinstance(body, str):
         body = body.splitlines()
-    yield assert_equal, body, block.lines
-    yield assert_is, glbs, block.glbs
-    yield assert_is, None, block.locs
+    assert body == block.lines
+    assert glbs is block.glbs
+    assert block.locs is None
```

Comment:
I think this is exactly that case. The asserts are all different from one another (and thus not parameterizable) but it would still be advantageous to have them all run.